tests_pagefetcher.py

```diff
@@ -0,0 +1,216 @@
+
+        with self.assertRaises(pagefetcher.TemplateNotAvailable):
+            pagefetcher.fetch("randomnumber")
+
+    def test_template_empty(self):
+
+        with self.assertRaises(pagefetcher.TemplateNotAvailable):
+            pagefetcher.fetch("")
+
+    def test_template_special_characters(self):
+
+        with self.assertRaises(pagefetcher.TemplateNotAvailable):
+            pagefetcher.fetch("fdsjkjl#@!#")
+
+    def test_template_int(self):
+
+        with self.assertRaises(pagefetcher.AurgumentIsNotAString):
+            pagefetcher.fetch(1)
+
+    def test_template_float(self):
+
+            with self.assertRaises(pagefetcher.AurgumentIsNotAString):
+                pagefetcher.fetch(2.4)
+
+    def test_template_empty_list(self):
+
+        template = []
+
+        with self.assertRaises(pagefetcher.AurgumentIsNotAString):
+            pagefetcher.fetch(template)
+
+    def test_template_non_empty_list(self):
+
+        template = [1, 2, 3]
+
+        with self.assertRaises(pagefetcher.AurgumentIsNotAString):
+            pagefetcher.fetch(template)
+
+    def test_template_empty_dict(self):
+
+        template = dict()
+
+        with self.assertRaises(pagefetcher.AurgumentIsNotAString):
+            pagefetcher.fetch(template)
+
+    def test_template_non_empty_dict(self):
+
+        template = {"test": 1, "test2": 2}
+
+        with self.assertRaises(pagefetcher.AurgumentIsNotAString):
+            pagefetcher.fetch(template)
+
+    def test_template_empty_set(self):
+
+        template = set()
+
+        with self.assertRaises(pagefetcher.AurgumentIsNotAString):
+            pagefetcher.fetch(template)
+
+    def test_template_non_empty_set(self):
+
+        template = {1, 2, 3}
+
+        with self.assertRaises(pagefetcher.AurgumentIsNotAString):
+            pagefetcher.fetch(template)
+
+    def test_template_None(self):
+
+        with self.assertRaises(pagefetcher.AurgumentIsNotAString):
+            pagefetcher.fetch(None)
+
+
+class TestDirectoryCheck(unittest.TestCase):
+    """ Tests directory_check function. """
+
+    def setUp(self):
+
+        os.makedirs("test")
+
+    def test_directory_exists(self):
+
+        path = "test/t1"
+        os.makedirs(path)
+        self.assertEqual(pagefetcher.directory_check(path), True,
+                         "Failed to find a correct directory!")
+        os.rmdir(path)
+
+    def test_directory_not_exists(self):
+
+        path = "test/t2"
+        self.assertEqual(pagefetcher.directory_check(path), False,
+                         "Failed, found a non existent directory!")
+
+    def test_directory_exists_with_space(self):
+
+        path = "test/t 2"
+        os.makedirs(path)
+        self.assertEqual(pagefetcher.directory_check(path), True,
+                         "Failed to find a correct directory with spaces!")
+        os.rmdir(path)
+
+    def tearDown(self):
+
+        os.rmdir("test")
+
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
Since this is our first unit test we should create a `tests` directory and put it inside.